test/src/Server/ODM/CRUD/CRUDTest.php

```diff
@@ -129,6 +132,46 @@ function (DoctrineResourceEvent $e) {
         $this->assertEquals('ZFTestCreateFailure', $body['detail']);
     }
 
+    /**
+     * @return void
+     */
+    public function testCreateByExplicitlySettingEntityFactoryInConstructor()
+    {
+        /** @var InstantiatorInterface|\PHPUnit_Framework_MockObject_MockObject $entityFactoryMock */
+        $entityFactoryMock = $this->getMockBuilder(InstantiatorInterface::class)->getMock();
+        $entityFactoryMock->expects(self::once())
+            ->method('instantiate')
+            ->with('ZFTestApigilityDbMongo\Document\Meta')
+            ->willReturnCallback(function ($class) {
+                return new $class;
+            });
+
+
+        /** @var ServiceManager $sm */
+        $sm = $this->getApplication()->getServiceManager();
+
+        $config = $sm->get('config');
+        $resourceName = 'ZFTestApigilityDbMongoApi\V1\Rest\Meta\MetaResource';
+        $resourceConfig = ! empty($config['zf-apigility']['doctrine-connected'][$resourceName])
+            ? $config['zf-apigility']['doctrine-connected'][$resourceName]
+            : [];
```

Comment:
Hm, why do we need it here? This is test, so we should know exactly what is our input data.